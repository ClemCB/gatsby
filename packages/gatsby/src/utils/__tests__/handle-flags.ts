import handleFlags from "../handle-flags"
import _ from "lodash"

describe(`handle flags`, () => {
  const activeFlags = [
    {
      name: `FAST_DEV`,
      env: `GATSBY_EXPERIMENTAL_FAST_DEV`,
      description: `Enable all experiments aimed at improving develop server start time`,
      includedFlags: [`DEV_SSR`, `QUERY_ON_DEMAND`],
    },
    {
      name: `DEV_SSR`,
      env: `GATSBY_EXPERIMENTAL_DEV_SSR`,
      description: `SSR pages on full reloads during develop. Helps you detect SSR bugs and fix them without needing to do full builds.`,
      umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/28138`,
    },
    {
      name: `QUERY_ON_DEMAND`,
      env: `GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND`,
      description: `Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.`,
      umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/27620`,
      noCi: true,
    },
  ]
  const configFlags = {
    FAST_DEV: true,
  }
  const configFlagsWithFalse = {
    QUERY_ON_DEMAND: true,
    DEV_SSR: false,
  }

  it(`returns validConfigFlags and a message`, () => {
    expect(handleFlags(activeFlags, configFlags)).toMatchSnapshot()
  })

  it(`filters out flags marked false`, () => {
    expect(
      handleFlags(activeFlags, configFlagsWithFalse).validConfigFlags
    ).toHaveLength(1)
  })
})
