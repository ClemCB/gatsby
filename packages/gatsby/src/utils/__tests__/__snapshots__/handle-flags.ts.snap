// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle flags returns validConfigFlags and a message 1`] = `
Object {
  "message": "The following flags are active:
- FAST_DEV - Enable all experiments aimed at improving develop server start time
- DEV_SSR (]8;;https://github.com/gatsbyjs/gatsby/discussions/28138Umbrella Issue]8;;) - SSR pages on full reloads during develop. Helps you detect SSR bugs and fix them without needing to do full builds.
- QUERY_ON_DEMAND (]8;;https://github.com/gatsbyjs/gatsby/discussions/27620Umbrella Issue]8;;) - Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.
",
  "validConfigFlags": Array [
    Object {
      "description": "Enable all experiments aimed at improving develop server start time",
      "env": "GATSBY_EXPERIMENTAL_FAST_DEV",
      "includedFlags": Array [
        "DEV_SSR",
        "QUERY_ON_DEMAND",
      ],
      "name": "FAST_DEV",
    },
    Object {
      "description": "SSR pages on full reloads during develop. Helps you detect SSR bugs and fix them without needing to do full builds.",
      "env": "GATSBY_EXPERIMENTAL_DEV_SSR",
      "name": "DEV_SSR",
      "umbrellaIssue": "https://github.com/gatsbyjs/gatsby/discussions/28138",
    },
    Object {
      "description": "Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.",
      "env": "GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND",
      "name": "QUERY_ON_DEMAND",
      "noCi": true,
      "umbrellaIssue": "https://github.com/gatsbyjs/gatsby/discussions/27620",
    },
  ],
}
`;
